insalan/tournament/views/substitute.py

```diff
@@ -0,0 +1,160 @@
+from django.core.exceptions import PermissionDenied, BadRequest
+from django.http import Http404
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth.hashers import check_password
+from django.http import QueryDict
+from django.contrib.auth.hashers import make_password
+
+from rest_framework import generics, permissions, status
+from rest_framework.exceptions import NotFound
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.authentication import SessionAuthentication
+
+from insalan.user.models import User, UserMailer
+import insalan.tournament.serializers as serializers
+
+from ..models import Player, Manager, Substitute, Event, Tournament, Game, Team, PaymentStatus
+from .permissions import ReadOnly, Patch
+
+
+class SubstituteRegistration(generics.RetrieveAPIView):
+    """Show a substitute registration"""
+
+    serializer_class = serializers.SubstituteSerializer
+    queryset = Substitute.objects.all().order_by("id")
+
+    def patch(self, request, *args, **kwargs):
+        """
+        Patch a substitute
+        """
+        user = request.user
+        data = request.data
+
+        if user is None or not user.is_authenticated:
+            raise PermissionDenied()
+
+        # get the substitute
+        substitute = Substitute.objects.get(id=kwargs["pk"])
+
+        # check if the user is related to the substitute
+        if substitute.as_user().id != user.id:
```

Comment:
just to be sure: we are supposing that the staff will use the django admin panel to patch an user and not the REST api?

Reply:
No, this code was only moved from the views.py to /views/. Theses endpoints are used to let players edit their registrations.